### PR TITLE
Api changes

### PR DESCRIPTION
**Pull request by jerrylinew**
Added several other functions the API needed to do, including updated comments for each of the API functions. Also included callbacks in database init function and moved database testing into callback so it does not crash on startup on a fresh database.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -90,16 +90,18 @@ app.get('/api/wish_list/get_users', wishListController.get_users);
...
  * Tests
  */
-const ownedBooksTest = require('./tests/models/owned_books').test();
-const possibleTradesTest = require('./tests/models/possible_trades').test();
-const usersTest = require('./tests/models/users').test();
+const initTest = require('./models/init').create_tables(function(){
+    const ownedBooksTest = require('./tests/models/owned_books').test();
+    const possibleTradesTest = require('./tests/models/possible_trades').test();
+    const usersTest = require('./tests/models/users').test();
+});
```
Comment (adiraju13) [inline app.js]: I am personally still not a fan of having our tests in app.js, I think we should create a separate file that people can run to run the tests

File: controllers/api/wish_list.js [hunk 2]
```diff
@@ -38,6 +42,22 @@ exports.remove_book = function(req, res) {
...
+            console.log("Successfully removed relation_want from the database!");
+
+        // res.json({status: status});
+    });
+
+    // delete all (A,B,C,D) in graph_edges where (A=user_id AND B=book_id) OR (C=user_id AND D=book_id)
+    ge.remove_wanted_book(user_id, book_id, function(status) {
+        if (status == ec.graph_edges_errors.DB_SUCCESS)
+            console.log("Successfully removed wanted book for all graph edges from the database!");
+
```
Comment (adiraju13) [inline controllers/api/wish_list.js:60]: Any error handling needed here?

File: controllers/api/possible_trades.js [hunk 3]
```diff
@@ -42,41 +65,46 @@ exports.remove_relation = function(req, res) {
...
-    var owned_book_id = req.body.owned_book_id;
 
-    db.remove_relation_have(user_id, owned_book_id, function(status){
+    ge.remove_user_owned_want(user_id, owned_book_id, wanted_book_id, function(status){
         if (status == ec.possible_trades_errors.DB_SUCCESS)
-            console.log("Successfully removed relation-have from the database!");
-
-        res.json({status: status});
+            console.log("Successfully removed edges from the database!");
     });
```
Comment (adiraju13) [inline controllers/api/possible_trades.js]: any error handling needed here?

**Review by adiraju13 — COMMENTED** [review]
There are a couple places where we might want to have more error handling, so maybe add that to the places I marked (if necessary, use your discretion). I also don't really see the tests for the code, but if the tests are in a different branch/already in master, and everything works, then it looks good to go.

File: controllers/api/wish_list.js (hunk 2 above)
Comment (karenl1) [reply controllers/api/wish_list.js:60]: Added test to check if wish list book isn't in the database

File: app.js (hunk 1 above)
Comment (jerrylinew) [reply app.js]: moved all testing to another file. Now just need to call one function in app.js to test everything

**Comment by adiraju13** [comment]
Look great! im going to merge

**Pull request merged**